src/main/java/org/eclipse/xpanse/terraform/boot/terraform/service/TerraformResultPersistenceManage.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ */
+
+package org.eclipse.xpanse.terraform.boot.terraform.service;
+
+import jakarta.annotation.Resource;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.UUID;
+import lombok.extern.slf4j.Slf4j;
+import org.eclipse.xpanse.terraform.boot.models.exceptions.FailedCallbackResponseStoreException;
+import org.eclipse.xpanse.terraform.boot.models.exceptions.ResultAlreadyReturnedOrRequestIdInvalidException;
+import org.eclipse.xpanse.terraform.boot.models.response.TerraformResult;
+import org.eclipse.xpanse.terraform.boot.utils.TerraformResultSerializer;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+/**
+ * Terraform service classes are manage task result.
+ */
+@Slf4j
+@Component
+public class TerraformResultPersistenceManage {
+
+    private static final String TF_RESULT_FILE_SUFFIX = ".dat";
+
+    @Value("${failed.callback.response.store.location}")
+    private String failedCallbackStoreLocation;
+
+    @Resource
+    private TerraformResultSerializer terraformResultSerializer;
+
+    /**
+     * When the terraform-boot callback fails, store the TerraformResult in the local file system.
+     *
+     * @param result TerraformResult.
+     */
+    public void persistTerraformResult(TerraformResult result) {
+        String filePath = getFilePath(result.getRequestId());
+        File file = new File(filePath);
+        if (!file.exists() && !file.mkdirs()) {
+            throw new FailedCallbackResponseStoreException("Failed to create directory: "
+                    + filePath);
+        }
+        byte[] terraformResultData = terraformResultSerializer.serialize(result);
+        try (FileOutputStream fos = new FileOutputStream(file.getPath() + File.separator
+                        + result.getRequestId() + TF_RESULT_FILE_SUFFIX)) {
+            fos.write(terraformResultData);
+            log.info("Store failed callback response success, fileName: {}",
```

Comment:
Log line must be terraform result successfully stored to 'filename' 
and the file name must be the name including the file extension. See if we can get the file name with the extension automatically using the eclipse serializer lib.